doc/neogit.txt

```diff
@@ -1848,6 +1848,13 @@ The following keys, in normal mode, will act on the commit under the cursor:
 • `b`    Insert breakpoint
 • `<cr>` Open current commit in Commit Buffer
 
+==============================================================================
+Custom Popups                                             *neogit_custom_popups*
+
+You can leverage Neogit's infrastructure to create your own popups and
+actions. Look at the built-in popups within `lua/neogit/popups/*` and the
+builder APIs in `lua/neogit/lib/popup/builder.lua` to get started.
+
```

Comment:
I think it would be helpful to provide a "hello world" type example, something that:
- Shows what to `require()`
- Shows how to bind the popup to a key via `vim.keymap.set()`
- Shows the switch/option/config/action/group methods (+ conditional variants)
- Shows how to bind a simple function to an action

The example can be _completely_ contrived; don't bother making it do anything actually useful. Sound reasonable?

Reply:
I've pushed an update that goes into a little more detail here. I wasn't sure how valuable trying to include _every_ one of the builder methods would be (and it would lead to a little more potential maintenance to keep things in sync). If you think that would be better I can add it, but I wonder if looking at the actual built-in popups would provide a more useful/realistic view of how to use them. Let me know!